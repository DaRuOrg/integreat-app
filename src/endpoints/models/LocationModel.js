import { isEmpty } from 'lodash/lang'

export default class LocationModel {
  constructor (name, path, live) {
    this._code = name
    this._name = path
    this._live = live
  }

  get live () {
    return this._live
  }

  get name () {
    return this._code
  }

  get path () {
    return this._name
  }

  get category () {
    return isEmpty(this._code) ? '?' : this._code[0].toUpperCase()
  }
}
