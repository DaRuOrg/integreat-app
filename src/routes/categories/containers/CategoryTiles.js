import React from 'react'
import PropTypes from 'prop-types'
import { Row } from 'react-flexbox-grid'

import PageModel from 'modules/endpoint/models/CategoryModel'
import LOCATIONS_ENDPOINT from 'modules/endpoint/endpoints/location'
import withFetcher from 'modules/endpoint/hocs/withFetcher'
import Caption from 'modules/common/components/Caption'
import CategoryTile from '../components/CategoryTile'

class CategoryTiles extends React.Component {
  static propTypes = {
    parentPage: PropTypes.instanceOf(PageModel).isRequired,
    pages: PropTypes.arrayOf(PropTypes.shape({
      page: PropTypes.instanceOf(PageModel).isRequired,
      url: PropTypes.string.isRequired
    })).isRequired
  }

  getTitle () {
    return this.props.locations.find((location) => location.code === this.props.parentPage.title).name
  }

  render () {
    return (
      <div>
        <Caption title={this.getTitle()}/>
        <Row>
          {this.props.pages.map(({page, url}) => <CategoryTile key={page.id} url={url} page={page}/>)}
        </Row>
      </div>
    )
  }
}

export default withFetcher(LOCATIONS_ENDPOINT, true, true)(CategoryTiles)
