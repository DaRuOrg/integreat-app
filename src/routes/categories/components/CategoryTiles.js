import React from 'react'
import PropTypes from 'prop-types'
import { Row } from 'react-flexbox-grid'

import CategoryModel from 'modules/endpoint/models/CategoryModel'
import Caption from 'modules/common/components/Caption'
import CategoryTile from './CategoryTile'
import LocationModel from '../../../modules/endpoint/models/LocationModel'

class CategoryTiles extends React.Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    categories: PropTypes.arrayOf(PropTypes.instanceOf(CategoryModel)).isRequired,
    locations: PropTypes.arrayOf(PropTypes.instanceOf(LocationModel)).isRequired
  }

  getTitle () {
    return this.props.locations.find((location) => location.code === this.props.title).name
  }

  render () {
    return (
      <div>
        <Caption title={this.getTitle()} />
        <Row>
          {this.props.categories.map(category =>
            <CategoryTile key={category.id}
                          category={category} />)}
        </Row>
      </div>
    )
  }
}

export default CategoryTiles
