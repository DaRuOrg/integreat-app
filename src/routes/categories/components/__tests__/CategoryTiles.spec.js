import React from 'react'
import { shallow } from 'enzyme'

import CategoryTiles from '../CategoryTiles'
import LocationModel from '../../../../modules/endpoint/models/LocationModel'

const locations = [
  new LocationModel({name: 'Augsburg', code: 'augsburg'}),
  new LocationModel({name: 'Stadt Regensburg', code: 'regensburg'}),
  new LocationModel({name: 'Werne', code: 'werne'})
]

const title = 'augsburg'

const categories = [
  {
    id: 35,
    url: '/augsburg/de/willkommen/willkommen-in-augsburg',
    title: 'Willkommen in Augsburg'
  },
  {
    id: 35,
    url: '/augsburg/de/willkommen/willkommen-in-augsburg',
    title: 'Willkommen in Augsburg'
  }
]

describe('CategoryTiles', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <CategoryTiles categories={categories}
                    locations={locations}
                    title={'augsburg'} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should get location names correctly', () => {
    const categoryTiles = shallow(
      <CategoryTiles categories={categories}
                     locations={locations}
                     title={'augsburg'} />
    ).instance()

    expect(categoryTiles.getLocationName(title)).toEqual('Augsburg')
    expect(categoryTiles.getLocationName('test')).toEqual('test')
  })
})
