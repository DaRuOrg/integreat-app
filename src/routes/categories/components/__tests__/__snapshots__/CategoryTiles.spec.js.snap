// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList should render 1`] = `
<div>
  <Caption
    title="Augsburg"
  />
  <Row
    className="categoryTiles"
  >
    <CategoryTile
      category={
        CategoryModel {
          "_availableLanguages": Object {},
          "_content": "",
          "_id": 35,
          "_order": 0,
          "_parentId": -1,
          "_parentUrl": null,
          "_thumbnail": null,
          "_title": "Willkommen in Augsburg",
          "_url": "/augsburg/de/willkommen/willkommen-in-augsburg",
        }
      }
      key="35"
    />
    <CategoryTile
      category={
        CategoryModel {
          "_availableLanguages": Object {},
          "_content": "",
          "_id": 35,
          "_order": 0,
          "_parentId": -1,
          "_parentUrl": null,
          "_thumbnail": null,
          "_title": "Willkommen in Augsburg",
          "_url": "/augsburg/de/willkommen/willkommen-in-augsburg",
        }
      }
      key="35"
    />
  </Row>
</div>
`;
